Setup PG, knex migrations, express and make health check route.

Skoryts opened the pull request.

Skoryts committed:
[#4] Setup PG, knex migrations, express and make health check route.

Skoryts committed:
[#4] Setup PG, knex migrations, express and make health check route.

belevtsev commented on api/app.js:
```diff
@@ -0,0 +1,9 @@
+const app = express();
+const port = 3000;
+
+require('./db');
```
You should create a db class and use it everywhere.
Please read how to use pool correctly.

belevtsev commented on api/app.js:
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const app = express();
+const port = 3000;
```
please use env file

belevtsev commented on api/db.js:
```diff
@@ -0,0 +1,13 @@
+const { Pool } = require('pg');
+const connectionString = 'postgres://wqhrrievdwrbco:4bc4d6ecd015e4a72356e19e9ddaebc8c0c343188716a94914489fd2af1be00c@ec2-54-217-236-201.eu-west-1.compute.amazonaws.com:5432/dfocldlmhfo197';
```
please use env file

belevtsev commented on api/db.js:
```diff
@@ -0,0 +1,13 @@
+  ssl: true
+});
+
+(async function(){
```
is it useful for you? If yes, please tell me why?

belevtsev commented on api/knexfile.js:
```diff
@@ -0,0 +1,35 @@
+    }
+  },
+
+  production: {
```
please remove

belevtsev commented on api/knexfile.js:
```diff
@@ -0,0 +1,35 @@
+
+  development: {
+    client: 'postgres',
+    connection: {
```
please use env file.

belevtsev commented on api/migrations/20180920142003_test.js:
```diff
@@ -0,0 +1,8 @@
+
+exports.up = function(knex, Promise) {
```
Please create something here.

Skoryts committed:
[#4] Make changes per comments in PR:
 - implement env files
 - create DB class
 - fix knexfile
 - make migration for base users table
 - enhance health check endpoint

belevtsev commented on .eslintrc.yml:
```diff
@@ -4,7 +4,6 @@ env:
 
 extends:
   - eslint:recommended
-  - airbnb
```
https://www.npmjs.com/package/eslint-config-airbnb-base

Skoryts replied:
I have installed, but now I need to left trailing comma! I hate trailing commas.

belevtsev replied:
let it be.

belevtsev commented on .eslintrc.yml:
```diff
@@ -16,9 +15,6 @@ rules:
   indent:
     - error
     - 2
-  linebreak-style:
```
linebreak-style should be present.

belevtsev commented on api/config.js:
```diff
@@ -0,0 +1,8 @@
+require('dotenv').config({path: '../.env'});
```
please add `{ silent: true }`.

belevtsev commented on api/config.js:
```diff
@@ -0,0 +1,8 @@
+
+const config = {
+  port: process.env.PORT,
+  db: process.env.DB_CONNECTION
```
`db ` -> `db_uri`

belevtsev commented on api/db.js:
```diff
@@ -0,0 +1,33 @@
+  }
+
+  query(query) {
+    return new Promise((resolve, reject) => {
```
async await

belevtsev commented on api/migrations/20180926201759_create_users.js:
```diff
@@ -0,0 +1,15 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('users', t => {
+    t.increments('id').unsigned().primary();
```
what about uuid?

belevtsev commented on api/migrations/20180926201759_create_users.js:
```diff
@@ -0,0 +1,15 @@
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('users', t => {
+    t.increments('id').unsigned().primary();
+    t.dateTime('createdAt').notNull();
```
default value?

belevtsev commented on api/migrations/20180926201759_create_users.js:
```diff
@@ -0,0 +1,15 @@
+  return knex.schema.createTable('users', t => {
+    t.increments('id').unsigned().primary();
+    t.dateTime('createdAt').notNull();
+    t.dateTime('updatedAt').nullable();
```
default value?

Skoryts replied:
Why we need default value? This column we need to fill, when row will be updated.

belevtsev replied:
You should have an experience to write triggers.

belevtsev commented on api/migrations/20180926201759_create_users.js:
```diff
@@ -0,0 +1,15 @@
+    t.increments('id').unsigned().primary();
+    t.dateTime('createdAt').notNull();
+    t.dateTime('updatedAt').nullable();
+    t.dateTime('deletedAt').nullable();
```
default value?

Skoryts replied:
Why we need default value? This column we need to fill, when row will be deleted.

belevtsev replied:
You should have an experience to write triggers.

belevtsev commented on .env.example:
```diff
@@ -0,0 +1,2 @@
+PORT=
+DB_CONNECTION=
```
`DB_CONNECTION ` -> `DB_URI`

Skoryts committed:
[#4] Make changes per comments in PR:
 - change migration with new uuid column for users
 - refactor health check row with async/await
 - change env vars with more readable
 - refactor method 'query' in DB class with async/await
 - add airbnb extends for eslint

Skoryts committed:
[#4] Create update function and trigger for users table
